examples/backoff.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: backoff-
+spec:
+  entrypoint: backoff
+  templates:
+  - name: backoff
+    backoffLimit: 4
+    container:
+      image: shrinand/random-fail
```

Comment:
The random-fail container takes an argument that is the probability of failure and fails with that probability. Here's the code: https://gist.github.com/shrinandj/7c3b49da704c5fb9406efd28178aa9e9

Reply:
Can we replace this with a script container to do this? I think it will make the example more explicit without the mystery of a what run.py does in the container.

Reply:
Script templates don't have retries as of now.